Tasks/ServiceFabricDeploy/Create-DiffPackage.psm1

```diff
@@ -0,0 +1,149 @@
+function Create-DiffPackage
+{
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory=$true)][String] $ApplicationName,
+        [parameter(Mandatory=$true)][String] $ApplicationPackagePath,
+        [parameter(Mandatory=$true)] $ConnectedServiceEndpoint,
+        [parameter(Mandatory=$true)][Hashtable] $ClusterConnectionParameters
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try {
+        Write-Host (Get-VstsLocString -Key DIFFPKG_CreatingDiffPackage)
+
+        Import-Module $PSScriptRoot\ps_modules\ServiceFabricHelpers
+        Import-Module $PSScriptRoot\ps_modules\PowershellHelpers
+
+        $appManifestName = "ApplicationManifest.xml"
+        $localAppManifestPath = Join-Path $ApplicationPackagePath $appManifestName
+        $localAppManifestXml = [XML](Get-Content $localAppManifestPath)
+        $applicationTypeName = $localAppManifestXml.ApplicationManifest.ApplicationTypeName
+        $localAppTypeVersion = $localAppManifestXml.ApplicationManifest.ApplicationTypeVersion
+
+        $app = Get-ServiceFabricApplication -ApplicationName $ApplicationName
+
+        # If $app is null, it means the application does not exist in the cluster. Diff Package is equal to Full Package. Use Full Package to do deployment
+        if (!$app -or $app.ApplicationTypeName -ne $applicationTypeName)
+        {
+            Write-Host (Get-VstsLocString -Key DIFFPKG_ApplicationDoesNotExist -ArgumentList @($ApplicationName, $ConnectedServiceEndpoint.Url))
+            Return
+        }
+
+        $diffPackagePath = Join-Path (Get-VstsInput -Name diffPackagePath -Require) "DiffPackage"
+        if (Test-Path -PathType Container -Path $diffPackagePath)
+        {
+            Remove-Item -Path $diffPackagePath -Recurse -Force
+        }
+        $diffPackagePath = New-Item -ItemType Directory -Path $diffPackagePath -Force
+        $diffPkgAppManifestPath = Join-Path $diffPackagePath $appManifestName
+
+        $clusterAppTypeVersion = $app.ApplicationTypeVersion
+
+        # If the ApplicationTypeVersion of the Application is not upgraded, no diff package is made because there is no need
+        if ($clusterAppTypeVersion -eq $localAppTypeVersion)
```

Comment:
This test can be moved right next to the one where app doesn't exist

Reply:
@sachinma Resolved